addons/mail/models/mail_mail.py

```diff
@@ -89,7 +89,7 @@ def default_get(self, fields):
         # protection for `default_type` values leaking from menu action context (e.g. for invoices)
         # To remove when automatic context propagation is removed in web client
         if self._context.get('default_type') not in type(self).message_type.base_field.selection:
-            self = self.with_context(dict(self._context, default_type=None))
+            self = self.with_context(dict(self._context, default_type=None, default_state='outgoing'))
```

Comment:
Would instead use the same condition as above for default_type

Reply:
I tried that at first, and all the tests became red in less than 5 minutes. 
I think it's because `{..., 'default_state': None`, ... } will still let the key 'default_state' in the context, so it will be used when creating. So unless we specifically remove the key from the `._context`, it will be considered not absent in the dict imo

Reply:
Found it @tde-banana-odoo https://github.com/odoo/odoo/commit/2bdf67d2df3

Reply:
Wanted to say something like
```
if self._context.get('default_type') not in type(self).message_type.base_field.selection:
    self = self.with_context(dict(self._context, default_type=None))
if self._context.get('default_state') not in type(self).state.base_field.selection:
    self = self.with_context(dict(self._context, default_state='outgoing'))
```

Reply:
Oh yes my bad I didn't see what you meant, sorry yeah I'll do that

Reply:
@tde-banana-odoo Changes applied, checks green, manual test okay, so if that's okay for you ? 
Also I think I also need someone from @odoo/rd-discuss but I know they're very busy atm :(

Reply:
This part is @tde-banana-odoo domain as it's pure mail, no need for us to check here, but thanks for asking.